examples/pruning/lightgbm_with_custom_cv_function.py

```diff
@@ -0,0 +1,404 @@
+
+    if MAKE_REPRODUCIBLE:
+        # turn off multiprocessing in Optuna samplers to ensure reproducible results
+        # (https://optuna.readthedocs.io/en/stable/faq.html#how-can-i-obtain-reproducible-optimization-results)
+        WORKERS_NUM = 1
+
+    # fix seed for data partitioning (train_test_split())
+    # and morel training (lgbm.fit())
+    SEED = 123
+
+    # select sampler for Optuna:
+    # (see more samplers at: https://optuna.readthedocs.io/en/stable/reference/samplers.html)
+    # - TPE (Tree-structured Parzen Estimator) sampling algo;
+    # note we fix sampler seed to make the sampling process deterministic
+    SAMPLER_SEED = 123
+    OPTUNA_SAMPLER = TPESampler(seed=SAMPLER_SEED)
+
+    # whether to prune unpromising trials to speed up studies
+    USE_PRUNING = True
+
+    # select pruner for Optuna:
+    # (see more pruners at: https://optuna.readthedocs.io/en/latest/reference/pruners.html)
+    # - median pruner
+    OPTUNA_PRUNER = MedianPruner()
+
+    # set verbosity level for lightgbm
+    VERBOSE_LGBM = -1
+
+    # Experiment
+
+    # Load data
+
+    train_x_df, _, train_y_df, _ = load_sklearn_toy_data(
+        objective=OBJECTIVE, test_size=0.01, as_df=True, seed=SEED
+    )
+
+    # Optimize params using Optuna
+
+    optuna_best_metrics = {}
+    optuna_best_params = {}
+
+    # Instantiate the custom function
+    objective = Objective_custom(
+        train_x_df,
+        train_y_df,
+        objective=OBJECTIVE,
+        eval_metric=METRIC,
+        eval_name=CV_EVAL_NAME,
+        folds=FOLDS,
+        fixed_rounds=FIXED_ROUNDS,
+        use_stopping=USE_STOPPING,
+        use_pruning=USE_PRUNING,
+        n_jobs=LGBM_NUM_THREADS,
+        seed=SEED,
+        verbosity=VERBOSE_LGBM,
+    )
+
+    study = optuna.create_study(sampler=OPTUNA_SAMPLER, pruner=OPTUNA_PRUNER, direction=OPT_DIR)
+
+    start_time = datetime.now()
+
+    # run Optuna optimization over the specified number of trials
+    study.optimize(objective, n_trials=TRIALS, n_jobs=WORKERS_NUM)
+
+    optim_time_custom = datetime.now() - start_time
+    print(
+        "\nOptuna+custom fun. study with %d trials took %.2f s. Time per trial: %.2f s."
+        % (TRIALS, optim_time_custom.total_seconds(), optim_time_custom.total_seconds() / TRIALS)
+    )
+
+    optuna_best_metrics["custom"] = study.best_value
+
+    best_trial = study.best_trial
+    optimized_best_params = best_trial.params
+
+    # append static parameters not returned by Optuna
+    static_params = {
+        "boosting_type": "gbdt",
+        "metric": METRIC,
+        "objective": OBJECTIVE,
+        "n_jobs": LGBM_NUM_THREADS,
+        "seed": SEED,
+        "verbosity": VERBOSE_LGBM,
+    }
+    all_best_params = {**optimized_best_params, **static_params}
+    optuna_best_params["custom"] = all_best_params
+
+    print(
+        "\nBest %s metric for optuna+custom fun. (CV models mean reported by Optuna): %.5f\n"
+        % (METRIC, optuna_best_metrics["custom"])
+    )
```

Comment:
Please use `format` instead of `%`.
```suggestion
    print(
        "\nBest {} metric for optuna+custom fun. (CV models mean reported by Optuna): {:.5f}\n".format(
            METRIC, optuna_best_metrics["custom"]
        )
    )
```

Reply:
I replaced `%`-style formatting with `{}`-style formatting also in [#L373](https://github.com/mirekphd/optuna/blob/lgbm-w-cust-cv-fun/examples/pruning/lightgbm_with_custom_cv_function.py#L373).